fv3core/stencils/updatedzc.py

```diff
@@ -220,44 +78,29 @@ def update_dz_c(
             xfx = p_weighted_average_bottom(ut, dp_ref)
             yfx = p_weighted_average_bottom(vt, dp_ref)
     with computation(PARALLEL), interval(...):
+        gz_x = gz
+        gz_x = corners.fill_corners_2cells_mult_x(gz_x, gz_x, 1.0, 1.0, 1.0, 1.0)
+        gz_y = gz_x
+        gz_y = corners.fill_corners_2cells_mult_y(gz_y, gz_y, 1.0, 1.0, 1.0, 1.0)
         fx, fy = xy_flux(gz_x, gz_y, xfx, yfx)
-        # TODO: check if below gz is ok, or if we need gz_y to pass this
-        gz = (gz_y * area + fx - fx[1, 0, 0] + fy - fy[0, 1, 0]) / (
-            area + xfx - xfx[1, 0, 0] + yfx - yfx[0, 1, 0]
-        )
+        # TODO: region for local validation only
+        with horizontal(
+            region[local_is - 1 : local_ie + 2, local_js - 1 : local_je + 2]
+        ):
+            gz = (gz_y * area + fx - fx[1, 0, 0] + fy - fy[0, 1, 0]) / (
+                area + xfx - xfx[1, 0, 0] + yfx - yfx[0, 1, 0]
+            )
     with computation(PARALLEL), interval(-1, None):
-        rdt = 1.0 / dt
-        ws3 = (zs - gz) * rdt
+        rdt = 1.0 / dt2
+        # TODO: region for local validation only
+        with horizontal(
+            region[local_is - 1 : local_ie + 2, local_js - 1 : local_je + 2]
+        ):
+            ws = (z_surface - gz) * rdt
     with computation(BACKWARD), interval(0, -1):
-        gz_kp1 = gz[0, 0, 1] + DZ_MIN
-        gz = gz if gz > gz_kp1 else gz_kp1
-
-
-def compute(dp_ref, zs, ut, vt, gz_in, ws3, dt2):
-    grid = spec.grid
-    origin = (1, 1, 0)
-    gz = copy(gz_in, origin=origin)
-    gz_x = copy(gz, origin=origin)
-    ws = copy(
-        ws3, origin=grid.full_origin(), domain=grid.domain_shape_full(add=(1, 1, 0))
-    )
-    corners.fill_corners_cells(gz_x, "x")
-    gz_y = copy(gz_x, origin=origin)
-    corners.fill_corners_cells(gz_y, "y")
-    update_dz_c(
-        dp_ref,
-        zs,
-        grid.area,
-        ut,
-        vt,
-        gz,
-        gz_x,
-        gz_y,
-        ws3,
-        dt=dt2,
-        origin=origin,
-        domain=(grid.nic + 3, grid.njc + 3, grid.npz + 1),
-    )
-    grid.overwrite_edges(gz, gz_in, 2, 2)
-    grid.overwrite_edges(ws3, ws, 2, 2)
-    return gz, ws3
+        # TODO region for local validation only
+        with horizontal(
+            region[local_is - 1 : local_ie + 2, local_js - 1 : local_je + 2]
+        ):
+            gz_kp1 = gz[0, 0, 1] + DZ_MIN
+            gz = gz if gz > gz_kp1 else gz_kp1
```

Comment:
Another local you could rename (`gz_kp1`), if you have a bit more context to add. Is this `gz_min` for the given height?

Reply:
good idea